Documentation/botr/readytorun-format.md

```diff
@@ -225,22 +236,67 @@ additional data determined by the flags.
 | ``READYTORUN_FIELD_SIG_MemberRefToken``      |  0x10 | If set, the token is memberref token. If not set, the token is fielddef token. |
 | ``READYTORUN_FIELD_SIG_OwnerType``           |  0x40 | Field type. Typespec appended as additional data. |
 
-### READYTORUN_ IMPORT_SECTIONS::AuxiliaryData
+### READYTORUN_IMPORT_SECTIONS::AuxiliaryData
 
 For slots resolved lazily via ``READYTORUN_HELPER_DelayLoad_MethodCall`` helper, auxiliary data are 
-compressed argument maps that allow precise GC stack scanning while the helper is running.
+compressed argument maps that allow precise GC stack scanning while the helper is running. The CoreCLR runtime class [``GCRefMapDecoder``](https://github.com/dotnet/coreclr/blob/6b9a3d3a87825b1a34bd8f114c9b181ce75b3b2e/src/inc/gcrefmap.h#L157) is used to parse this information. This data would not be required for runtimes that allow conservative stack scanning.
+
+The auxiliary data table is supposed to contain the exact same number of GC ref map records as there are method entries in the import section. To accelerate GC ref map lookup, the auxiliary data section starts with a lookup table holding the offset of every 1024-th method in the runtime function table within the linearized GC ref map.
+
+| Offset in auxiliary data | Size | Content |
+|-------------------------:|-----:|:--------|
+| 0 | 4 | Offset to GC ref map info for method #0 relative to this byte i.e. 4 * (MethodCount / 1024 + 1) |
+| 4 | 4 | Offset to GC ref map info for method #1024 |
+| 8 | 4 | Offset to GC ref map info for method #2048 |
+| ... | | |
+4 * (MethodCount / 1024 + 1) | ... | Serialized GC ref map info |
+
+The GCRef map is used to encode GC type of arguments for callsites. Logically, it is a sequence ``<pos, token>`` where ``pos`` is 
+position of the reference in the stack frame and ``token`` is type of GC reference (one of [``GCREFMAP_XXX``](https://github.com/dotnet/coreclr/blob/6b9a3d3a87825b1a34bd8f114c9b181ce75b3b2e/src/inc/corcompile.h#L633) values):
+
+| CORCOMPILE_GCREFMAP_TOKENS | Value | Stack frame entry interpretation |
+|:---------------------------|------:|:--------|
+| ``GCREFMAP_SKIP`` | 0 | Not a GC-relevant entry |
+| ``GCREFMAP_REF`` | 1 | GC reference |
+| ``GCREFMAP_INTERIOR`` | 2 | Pointer to a GC reference |
+| ``GCREFMAP_METHOD_PARAM`` | 3 | Hidden method instantiation argument to generic method |
+| ``GCREFMAP_TYPE_PARAM`` | 4 | Hidden type instantiation argument to generic method |
+| ``GCREFMAP_VASIG_COOKIE`` | 5 | VARARG signature cookie |
+
+The position values are calculated in ``size_t`` aka ``IntPtr`` units (4 bytes for 32-bit architectures vs. 8 bytes for 64-bit architectures) and start at the offset ``TransitionBlock::GetOffsetOfFirstGCRefMapSlot`` relative to the transition frame address. For x86 the position encoding is [somewhat more complicated](https://github.com/dotnet/coreclr/blob/d5d18896900561b7aaf38ba9501a8525a4b9caea/src/vm/frames.cpp#L1326).
```

Comment:
This should not be referencing the implementation. 

The position is a index into a logical array that starts with spilled argument registers, and continues with stack registers. This is true for both x86 and non-x86, I think.

Reply:
Well, it's just that on x86 the spilled argument registers are indexed in reverse order (or perhaps they are somehow spilled in reverse order, I actually don't know what's the exact problem there). I can easily remove the reference to the implementation if you find it inappropriate though.

Reply:
Not addressed in 2nd commit - waiting on your further feedback.

Reply:
>  I can easily remove the reference to the implementation if you find it inappropriate though.

Yes, I think it would be best to not reference the implementation here.

Reply:
Removed in 3rd commit.